providers/github/resources/github_repo.go

```diff
@@ -470,8 +470,26 @@ func (g *mqlGithubBranch) protectionRules() (*mqlGithubBranchprotection, error)
 	branchProtection, _, err := conn.Client().Repositories.GetBranchProtection(context.TODO(), ownerName, repoName, branchName)
 	if err != nil {
 		// NOTE it is possible that the branch does not have any protection rules, therefore we don't return an error
+		if strings.Contains(err.Error(), "Not Found") {
+			res, err := CreateResource(g.MqlRuntime, "github.branchprotection", map[string]*llx.RawData{
+				"id":                             llx.StringData(repoName + "/" + branchName),
+				"requiredStatusChecks":           llx.MapData(map[string]interface{}{}, types.Any),
+				"requiredPullRequestReviews":     llx.MapData(map[string]interface{}{}, types.Any),
+				"enforceAdmins":                  llx.MapData(map[string]interface{}{}, types.Any),
+				"restrictions":                   llx.MapData(map[string]interface{}{}, types.Any),
+				"requireLinearHistory":           llx.MapData(map[string]interface{}{}, types.Any),
+				"allowForcePushes":               llx.MapData(map[string]interface{}{}, types.Any),
+				"allowDeletions":                 llx.MapData(map[string]interface{}{}, types.Any),
+				"requiredConversationResolution": llx.MapData(map[string]interface{}{}, types.Any),
+				"requiredSignatures":             llx.BoolData(false),
+			})
+			if err != nil {
+				return nil, err
+			}
+			return res.(*mqlGithubBranchprotection), nil
+		}
 		// TODO: figure out if the client has the permission to fetch the protection rules
-		return nil, nil
+		return nil, err
```

Comment:
@vjeffrey I'd like your opinion on this one.

When I keep it as is, this panics:
```
cnquery run github repo czunker/go-vex -c "github.repository{ name branches[0]{*} commits[0]{*} }" 
! using builtin provider for github
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
! using builtin provider for github
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1180360]

goroutine 14 [running]:
go.mondoo.com/cnquery/providers/github/resources.(*mqlGithubBranchprotection).MqlID(0x2f0000c000cd04ea?)
	/home/christian/workspace/mondoo/github.com/cnquery/providers/github/resources/github.lr.go:3937
go.mondoo.com/cnquery/llx.resource2result({0x22861a0, 0x0}, {0xc000fea348, 0x18})
	/home/christian/workspace/mondoo/github.com/cnquery/llx/data_conversions.go:313 +0x52
```

When I only create a resource with an ID, it prevents the panic, but I get "cannot convert" error.

In v8, this simply displayed as "null":
```
cnquery run github repo czunker/go-vex -c "github.repository{ name branches[0]{ name protectionRules {*} } commits[0]{ sha commit {*}} }"
...
github.repository: {
...
  name: "go-vex"
  branches[0]: {
    protectionRules: null
    name: "christian/extend_csaf_advisories"
  }
}
```

This change works, but is quite different from v8. Also, this might apply to other resources.

Reply:
i'll check that out, i think i ran into that before on aws

Reply:
this looks like it's an issue not limited to this resource -- returning nil in these types of situations isn't behaving the way we expect it to - so we can probably move this pr forward and fix that in a followup

Reply:
I rebased this on main, as https://github.com/mondoohq/cnquery/pull/1726 was merged.

I changed the field accordingly, and now it looks good:
```
nquery run github repo czunker/go-vex -c "github.repository{ name branches[0]{ name protectionRules {*} }}" --verbose 
DBG using provider github with connector github
DBG no need to update provider last-refresh=29.146104804s provider=github
DBG running provider plugin path=/home/christian/.config/mondoo/providers/github/github
→ loaded configuration from /etc/opt/mondoo/mondoo.yml using source default
DBG resolved 1 assets
...
github.repository: {
  name: "go-vex"
  branches[0]: {
    protectionRules: null
    name: "christian/extend_csaf_advisories"
  }
}
```

Reply:
awesome! thank you @arlimus for that fix!